src/util/branch-name.ts

```diff
@@ -103,7 +103,22 @@ const AUTORELEASE_PATTERN =
   /^release-?(?<component>[\w-.]*)?-v(?<version>[0-9].*)$/;
 class AutoreleaseBranchName extends BranchName {
   static matches(branchName: string): boolean {
-    return !!branchName.match(AUTORELEASE_PATTERN);
+    const match = branchName.match(AUTORELEASE_PATTERN);
+    if (!match) {
+      return false;
+    }
+
+    // The regex can match release branches that should belong to
+    // DefaultBranchName (such as release-please--branches--v3.3.x). Make sure
+    // the component, if present, isn't something suspicious. See also
+    // https://github.com/googleapis/release-please/issues/1310
+    if (match?.groups) {
+      const component = match.groups['component'];
```

Comment:
```suggestion
      const component = match.groups.component
```

stylistic-nit.

Reply:
Thanks, I'm new to TS.  Still doing JS + Closure in my own projects.